test/Tests/src/Data/Json_Spec.enso

```diff
@@ -106,66 +84,47 @@ spec =
+            object.get "foo" . should_equal "bar"
+            object.get "bar" . should_equal Nothing
+            object.get "bar" 1 . should_equal 1
+            object.at "foo" . should_equal "bar"
+            object.at "bar" . should_fail_with No_Such_Key.Error
+            Json.parse "null" . get "foo" . should_equal Nothing
+            Json.parse "null" . get "foo" 1 . should_equal 1
 
         Test.specify "should be able to get field_names" <|
-            Json.parse '{ "foo": "bar", "baz": ["foo", "x", false] }' . field_names . should_equal ["baz", "foo"]
+            Json.parse '{ "foo": "bar", "baz": ["foo", "x", false] }' . field_names . should_equal ["foo", "baz"]
             Json.parse '{}' . field_names . should_equal []
-            Json.parse 'null' . field_names . should_fail_with Illegal_Argument.Error
-            Json.parse '[1,2,3]' . field_names . should_fail_with Illegal_Argument.Error
 
         Test.specify "should be able to get a value by index" <|
-            Json.parse "[1, 2, 3, 4, 5]" . at 2 . should_equal (Json.Number 3)
-            Json.parse "[1, 2, 3, 4, 5]" . at -2 . should_equal (Json.Number 4)
+            Json.parse "[1, 2, 3, 4, 5]" . at 2 . should_equal 3
+            Json.parse "[1, 2, 3, 4, 5]" . at -2 . should_equal 4
             Json.parse "[1, 2, 3, 4, 5]" . at 5 . should_fail_with Index_Out_Of_Bounds.Error
-            Json.parse '"foo"' . at 0 . should_equal (Json.String "f")
-            Json.parse '"foo"' . at -1 . should_equal (Json.String "o")
-            Json.parse '"foo"' . at 3 . should_fail_with Index_Out_Of_Bounds.Error
-            Json.parse '{}' . at 1 . should_fail_with Illegal_Argument.Error
-
-        Test.specify "should be able to make empty objects and array" <|
-            Json.empty_object.should_equal (Json.Object Map.empty)
-            Json.empty_object.to_text.should_equal "{}"
-            Json.empty_array.should_equal (Json.Array [])
-            Json.empty_array.to_text.should_equal "[]"
+            Json.parse "[1, 2, 3, 4, 5]" . get 2 . should_equal 3
+            Json.parse "[1, 2, 3, 4, 5]" . get -2 . should_equal 4
+            Json.parse "[1, 2, 3, 4, 5]" . get 5 . should_equal Nothing
+            Json.parse "[1, 2, 3, 4, 5]" . get 5 "?" . should_equal "?"
 
         Test.specify "should be able to get length" <|
-            Json.empty_object.length.should_equal 0
-            Json.empty_array.length.should_equal 0
+            Json.parse "{}" . length . should_equal 0
+            Json.parse "[]" . length . should_equal 0
             Json.parse '{ "foo": "bar", "baz": ["foo", "x", false] }' . length . should_equal 2
             Json.parse '[1, 2, 3, 4, 5]' . length . should_equal 5
             Json.parse '"foo"' . length . should_equal 3
             Json.parse '""' . length . should_equal 0
-            Json.Null.length.should_fail_with Illegal_Argument.Error
 
```

Comment:
I'd add a test here like
```
obj1 = JS_Object.from_pairs [["a", 42], ["b", "C"]]
obj2 = JS_Object.from_pairs [["a", 42], ["b", "C"]]
obj3 = JS_Object.from_pairs [["a", 42], ["D", "C"]]

obj1.should_equal obj2
obj1.should_not_equal obj3
```

We rely in our tests quite a lot on JS Object equality - so good to have some sanity check - especially one that `a == b` is not _always_ true for JS Objects. It probably isn't but it's possible to have a bug so a sanity check could be good.